cmd/libsnap-confine-private/snap.c

```diff
@@ -168,3 +168,24 @@ void sc_snap_name_validate(const char *snap_name, struct sc_error **errorp)
  out:
 	sc_error_forward(errorp, err);
 }
+
+void sc_snap_drop_instance_name(const char *snap_name, char *base, size_t base_len)
+{
+	if (snap_name == NULL || base == NULL) {
+		die("base or snap name unset");
+	}
+
+	const char *pos = strchr(snap_name, '_');
+	size_t blen = 0;
+	if (pos == NULL) {
+		blen = strlen(snap_name);
+	} else {
+		blen = (pos - snap_name);
+	}
+
+	if (blen > base_len) {
+		die("base buffer too small");
+	}
```

Comment:
Off-by-one. If snap_name is 'foo', then blen is '3'. If base_len is 3, then you have a memcpy that sets base as foo but there is no space to NUL terminate.